source/extensions/grpc_credentials/file_based_metadata/config.cc

```diff
@@ -27,9 +27,10 @@ FileBasedMetadataGrpcCredentialsFactory::getChannelCredentials(
     case envoy::api::v2::core::GrpcService::GoogleGrpc::CallCredentials::kFromPlugin: {
       if (credential.from_plugin().name() == GrpcCredentialsNames::get().FileBasedMetadata) {
         FileBasedMetadataGrpcCredentialsFactory file_based_metadata_credentials_factory;
+        // downcast to file based config, ignore unknown fields (kuat)
         const Envoy::ProtobufTypes::MessagePtr file_based_metadata_config_message =
             Envoy::Config::Utility::translateToFactoryConfig(
-                credential.from_plugin(), file_based_metadata_credentials_factory);
+                credential.from_plugin(), file_based_metadata_credentials_factory, true);
```

Comment:
Maybe I'm brain farting, but can you reiterate why we need to do internal conversions with lenient settings, perhaps more lenient than the external ones? I remember there was some reason, but it would be helpful to know this.

Also, I think that having this boolean is a one of those needles/poison needles minefields; people will have trouble reasoning when writing code when to set it to true vs. false. We should probably have two versions of `translateToFactoryConfig` that have names that reflect their intended use, e.g. internal vs. external or whatever.

Reply:
I think this needs to go case-by-case basis. If the source is a proto, then being lenient is fine. If the source is a struct, then we should ideally follow the same conventions. I don't have a good model of all the places we need this translation except for filter configs. And it started to get way more invasive to insert this boolean than I wished for.

Reply:
We will need to update the style guide to make this very clear. There's really multiple risks; most contributors won't understand this distinction clearly and even reviewers are likely to not fully grok this or apply inconsistently. Semantics naming of the conversion utils to reflect the different expected use cases seems safer where possible, or anything we can do to structurally make it harder to reach up and grab the poison tipped needles.

Reply:
Also, what do we gain by being lenient with proto? Presumably we've already vetted for no unknown fields at the edge of config ingest, so we should not hit any unknown fields during internal conversions?